erpnext/accounts/report/budget_variance_report/budget_variance_report.py

```diff
@@ -2,187 +2,419 @@
+        """
+			select
+				gl.account,
+				gl.debit,
+				gl.credit,
+				gl.fiscal_year,
+				MONTHNAME(gl.posting_date) as month_name,
+				b.{budget_against} as budget_against
+			from
+				`tabGL Entry` gl,
+				`tabBudget Account` ba,
+				`tabBudget` b
+			where
+				b.name = ba.parent
+				and b.docstatus = 1
+				and ba.account=gl.account
+				and b.{budget_against} = gl.{budget_against}
+				and gl.fiscal_year between %s and %s
+				and b.{budget_against} = %s
+				and exists(
+					select
+						name
+					from
+						`tab{tab}`
+					where
+						name = gl.{budget_against}
+						{cond}
+				)
+				group by
+					gl.name
+				order by gl.fiscal_year
+		""".format(
+            tab=filters.budget_against, budget_against=budget_against, cond=cond
+        ),
+        (filters.from_fiscal_year, filters.to_fiscal_year, name),
+        as_dict=1,
+    )
+
+    cc_actual_details = {}
+    for d in ac_details:
+        cc_actual_details.setdefault(d.account, []).append(d)
+
+    return cc_actual_details
+
 
 def get_dimension_account_month_map(filters):
-	import datetime
-	dimension_target_details = get_dimension_target_details(filters)
-	tdd = get_target_distribution_details(filters)
+    dimension_target_details = get_dimension_target_details(filters)
+    tdd = get_target_distribution_details(filters)
 
-	cam_map = {}
+    cam_map = {}
 
-	for ccd in dimension_target_details:
-		actual_details = get_actual_details(ccd.budget_against, filters)
+    for ccd in dimension_target_details:
+        accounts = get_accounts(ccd.name, filters)
+        actual_details = get_actual_details(ccd.name, filters)
 
-		for month_id in range(1, 13):
-			month = datetime.date(2013, month_id, 1).strftime('%B')
-			cam_map.setdefault(ccd.budget_against, {}).setdefault(ccd.account, {}).setdefault(ccd.fiscal_year,{})\
-				.setdefault(month, frappe._dict({
-					"target": 0.0, "actual": 0.0
-				}))
+        for year in get_fiscal_years(filters):
+            year = year[0]
+            monthly_distribution = get_monthly_distribution(ccd.name, year, filters)
+            for month_id in range(1, 13):
+                month = datetime.date(2013, month_id, 1).strftime(
+                    "%B"
+                )  # Get month string
 
-			tav_dict = cam_map[ccd.budget_against][ccd.account][ccd.fiscal_year][month]
-			month_percentage = tdd.get(ccd.monthly_distribution, {}).get(month, 0) \
-				if ccd.monthly_distribution else 100.0/12
+                for account in accounts:
+                    account = account[0]
+                    cam_map.setdefault(ccd.name, {}).setdefault(account, {}).setdefault(
+                        year, {}
+                    ).setdefault(month, frappe._dict({"target": 0.0, "actual": 0.0}))
 
-			tav_dict.target = flt(ccd.budget_amount) * month_percentage / 100
+                    tav_dict = cam_map[ccd.name][account][year][month]
 
-			for ad in actual_details.get(ccd.account, []):
-				if ad.month_name == month:
```

Comment:
@kevingdc No need to make all these changes, a small change here would have fixed the issue. No need make extra queries to get budget_amount or monthly_distribution amount seperately.

`if ad.month == month and ad.fiscal_year == ccd.fiscal_year` this condition would have been enough. Also there are lot of inconsistencies in indentation spacing, somewhere spaces and somewhere tabs, we usually prefer tabs and that is what we have used in the entire project

Reply:
Hi @deepeshgarg007,

Thank you for the response. I appreciate the feedback, and you're right, adding the the condition `if ad.month == month and ad.fiscal_year == ccd.fiscal_year` would have fixed it. Although the reason why I added those functions was so I could also show the actual expense even though it doesn't have a budget allocated, an example is shown below:

<img width="1179" alt="Screen Shot 2020-05-08 at 4 55 26 PM" src="https://user-images.githubusercontent.com/16395892/81390049-674f9700-914d-11ea-87ef-7fb4d8a36c51.png">

The old code, even though it would get fixed by adding the condition you mentioned wouldn't have shown it. The output would have been like this:

<img width="1177" alt="Screen Shot 2020-05-08 at 5 00 58 PM" src="https://user-images.githubusercontent.com/16395892/81390111-80f0de80-914d-11ea-99d7-8774152ef2c4.png">


I thought it would be more helpful to the users to see their actual expenses even though there wasn't a budget allocated for it. But if you feel that this is unnecessary, please tell me and I'll update my code to reflect your request. 

Thank you!

Reply:
Budget Variance Report is to check variation of Actual Expense from Budget. This doesn't seems the right place to show expense if budget doesn't exits, also it adds a lot of unnecessary query to get this done and fix the spaces for indentation as welll

Reply:
@deepeshgarg007 got it. Updated the code to reflect this.